extensions/amp-carousel/0.1/slidescroll.js

```diff
@@ -159,6 +159,17 @@ export class AmpSlideScroll extends BaseSlides {
       this.slidesContainer_.addEventListener(
           'touchend', this.touchEndHandler_.bind(this));
     }
+
+    this.registerAction('moveToSlideIndex', invocation => {
```

Comment:
amp-carousel does not have any on: events that are supported - i like your first option.